File: Cs/Solution/DeleteGreatestValue.cs

namespace Solution
{
	public partial class Solution
	{
		public int DeleteGreatestValue(int[][] grid)
		{
			foreach (int[] row in grid)
			{
				System.Array.Sort(row);
			}
			int ret = 0;
			for (int j = 0; j < grid[0].Length; j++)
			{
				int max = 0;
				for (int i = 0; i < grid.Length; i++)
				{
					max = System.Math.Max(max, grid[i][j]);
				}
				ret += max;
			}
			return ret;
		}
	}
}

File: Ruby/solution/delete_greatest_value.rb
# @param {Integer[][]} grid
# @return {Integer}
def delete_greatest_value(grid)
	grid.each(&:sort!).transpose.map(&:max).sum
end

input0 = [[[1,2,4],[3,3,1]], [[10]]]
expect = [8, 10]

fail_cnt = 0
pass_cnt = 0

puts "\n"
for i in 0...input0.length
        output = delete_greatest_value(input0[i])
        if expect[i] != output
                fail_cnt += 1
                puts "Input #{i} expect #{expect[i]}, actual #{output}"
        else
                pass_cnt += 1
        end
end

puts "==== Test Summary ===="
if fail_cnt == 0
        puts "TEST PASS!"
else
        puts "TEST FAIL!"
end
puts "test #{fail_cnt+pass_cnt}"
puts "pass #{pass_cnt}"
puts "fail #{fail_cnt}"
puts "======================"
